### PR TITLE
fix(controller): make tag selector ignore references in an oci image index that don't look like they refer to an image

### LINKED ISSUE #1267: Warehouse can't watch images: cannot parse "" as "2006"'
# Description

Recently my Warehouse stopped producing new freight. The error is:

```
status:
  error: 'error getting latest Freight from repositories: error syncing image repo
    subscriptions: error getting latest suitable tag for image "ghcr.io/jessesuen/guestbook":
    error fetching newest applicable tag for image "ghcr.io/jessesuen/guestbook":
    error retrieving tag "v0.0.47": error extracting tag from manifest "v0.0.47":
    error getting tag from manifest sha256:96d355d029898e9157cdd178a9c78b8b00496ce868f64a1a107fa7dba568c7ad:
    error extracting tag from manifest sha256:96d355d029898e9157cdd178a9c78b8b00496ce868f64a1a107fa7dba568c7ad:
    error parsing createdAt timestamp from blob sha256:9d752d900567cbd9197b95b1847c921c5ca04b0d3380c73da3c83aefe51dd4f6
    referenced by OCI manifest sha256:96d355d029898e9157cdd178a9c78b8b00496ce868f64a1a107fa7dba568c7ad:
    parsing time "" as "2006-01-02T15:04:05.999999999Z07:00": cannot parse "" as "2006"'
```

# Steps to Reproduce

Try subscribing to: `ghcr.io/jessesuen/guestbook`

# Version

tip as of today

# Logs

```
Paste any relevant application logs here.
```

### PR DESCRIPTION
**Pull request by krancour**
Fixes #1267

### CODE REVIEW

File: internal/image/tag_selector_ghcr_test.go [hunk 1]
```diff
@@ -18,7 +18,8 @@ import (
 // may get rate-limited executing these tests, so they're disabled by default.
 
 func TestSelectTagGHCR(t *testing.T) {
-	const kargoRepo = "ghcr.io/akuity/kargo"
+	const guestbookRepo = "ghcr.io/jessesuen/guestbook"
+	const guestbookTag = "v0.0.47"
```
Comment (jessesuen) [inline internal/image/tag_selector_ghcr_test.go:22]: Hmm, I'd like the freedom to delete this repo without breaking tests. I've repushed this multi-arch image as: `ghcr.io/akuity/kargo-test:unknown`. Can we change this to use that?

https://github.com/orgs/akuity/packages/container/kargo-test/156938778?tag=unknown

**Review by jessesuen — APPROVED** [review]
Pre-emptive approval. As discussed, you can use `ghcr.io/akuity/kargo-test:unknown` for testing the edge case.
